src/server/util/aws.ts

```diff
@@ -0,0 +1,41 @@
+import { parse, URL } from 'url'
+import { S3 } from 'aws-sdk'
+import { s3Bucket } from '../config'
+
+export const s3 = new S3()
+
+/**
+ * Reformat the pre-signed url to one that will be accepted by S3.
+ *
+ * This is necessary because we used a gov.sg CNAME and configured S3
+ * to only accept requests that have that gov.sg CNAME as the path.
+ *
+ * Broadly speaking, we need to change https://s3.amazonaws.com/file.go.gov.sg/link?search
+ * to https://file.go.gov.sg.s3.amazonaws.com/link?search.
+ */
+const reformatPresignedUrl = (url: string, fileName: string) => {
+  const urlObj = parse(url)
+  const {
+    host,
+    pathname,
+    protocol,
+    search,
+  } = urlObj
+  const newUrl = new URL('https://lorem-ipsum.com')
+  newUrl.protocol = protocol as string
+  newUrl.host = `${pathname?.split('/')[1]}.${host}`
```

Comment:
Any way to avoid doing a string split?

Reply:
frankly speaking i'm not too sure if there are better ways to do this, it's just what i came up with. the goal is to extract the `file[-staging].go.gov.sg` from `/file[-staging].go.gov.sg/<optional directories>/filename`